### PR TITLE
feat: Add version information

### PR DESCRIPTION
**Pull request by joekr**
**What this PR does / why we need it**:
This allows us to use build version information to help debug
issues.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
No open issue

### CODE REVIEW

File: docs/src/SUMMARY.md [hunk 1]
```diff
@@ -19,7 +19,7 @@
   - [Install Cluster API for Oracle Cloud Infrastructure](./gs/install-cluster-api.md)
   - [Create Workload Cluster](./gs/create-workload-cluster.md)
   - [Create Workload Templates](./gs/create-workload-templates.md)
-  - [Using externally managed infrastructure](./gs/externally-managed-infrastructure.md)
+  - [Using externally managed infrastructure](./gs/externally-managed-cluster-infrastructure.md)
```
Comment (joekr) [inline docs/src/SUMMARY.md:22]: These files shouldn't be changed. I'll have to look into why they are being updated.

**Comment by joekr** [comment]
# unit tests

```
?       github.com/oracle/cluster-api-provider-oci      [no test files]
ok      github.com/oracle/cluster-api-provider-oci/api/v1beta1  1.303s  coverage: 18.7% of statements
?       github.com/oracle/cluster-api-provider-oci/cloud/config [no test files]
ok      github.com/oracle/cluster-api-provider-oci/cloud/ociutil        0.711s  coverage: 25.0% of statements
ok      github.com/oracle/cluster-api-provider-oci/cloud/scope  177.792s        coverage: 73.5% of statements
?       github.com/oracle/cluster-api-provider-oci/cloud/scope/mocks    [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/compute       [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/compute/mock_compute  [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement     [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/identity      [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/identity/mock_identity        [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer   [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer/mock_nlb  [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/vcn   [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/vcn/mock_vcn  [no test files]
ok      github.com/oracle/cluster-api-provider-oci/controllers  27.591s coverage: 71.0% of statements
?       github.com/oracle/cluster-api-provider-oci/exp/api/v1beta1      [no test files]
?       github.com/oracle/cluster-api-provider-oci/exp/controllers      [no test files]
?       github.com/oracle/cluster-api-provider-oci/feature      [no test files]
?       github.com/oracle/cluster-api-provider-oci/version      [no test files]
```

**Review by shyamradhakrishnan — APPROVED** [review]
can you please add a comment on the version will look like? just an example

**Comment by joekr** [comment]
Version examples 

- `development` - if the `GitVersion` isn't set
- `v0.4.0-dirty` - on a working branch if there are changes
- `v0.4.0` - on the release once the release is actually built (this is based on the git tag)

**Pull request merged**